### PR TITLE
add tidb permit host option

### PR DESCRIPTION
**Pull request by shonge**
### What problem does this PR solve? 
#733
### What is changed and how does it work?
Add an option permitHost in values, and add the logic in _initialize_tidb_users.py.tpl  
### Check List 

Tests 

 - Manual test

Code changes

 - Has Helm charts change

Side effects

None

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
Users can designate permitting host in chart values.tidb.permitHost
 ```

### CODE REVIEW

File: charts/tidb-cluster/values.yaml [hunk 1]
```diff
@@ -264,6 +264,9 @@ tidb:
   # kubectl create secret generic tidb-secret --from-literal=root=<root-password> --namespace=<namespace>
   # If unset, the root password will be empty and you can set it after connecting
   # passwordSecretName: tidb-secret
+  # permitHost is the host which will only be allowed to connect to the TiDB.
+  # If unset, defaults to '%'.
```
Comment (tennix) [inline charts/tidb-cluster/values.yaml]:
```suggestion
  # If unset, defaults to '%' which means allow any host to connect to the TiDB.
```

File: charts/tidb-cluster/templates/scripts/_initialize_tidb_users.py.tpl [hunk 2]
```diff
@@ -1,5 +1,10 @@
 import os, MySQLdb
 host = '{{ template "cluster.name" . }}-tidb'
+{{- if .Values.tidb.permitHost }}
+permit_host = '{{ .Values.tidb.permitHost }}'
```
Comment (tennix) [inline charts/tidb-cluster/templates/scripts/_initialize_tidb_users.py.tpl]:
```suggestion
permit_host = {{ .Values.tidb.permitHost | default %% | quote }}
```

File: charts/tidb-cluster/templates/tidb-initializer-job.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{{- if or .Values.tidb.passwordSecretName .Values.tidb.initSql }}
+{{- if or .Values.tidb.permitHost .Values.tidb.passwordSecretName .Values.tidb.initSql }}
```
Comment (tennix) [inline charts/tidb-cluster/templates/tidb-initializer-job.yaml]: I think the initializer cannot work until the `passwordSecretName` is set. So the `permitHost` should be after `passwordSecretName`.

**Comment by weekface** [comment]
@shonge The e2e failed, can't set password to the tidb cluster

```
$ kubectl logs -f -n e2e cluster4-tidb-initializer-kfwbt
Traceback (most recent call last):
  File "<string>", line 18, in <module>
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 206, in execute
    res = self._query(query)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 312, in _query
    db.query(q)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/connections.py", line 224, in query
    _mysql.connection.query(self, query)
MySQLdb._exceptions.OperationalError: (1133, "Can't find any matching row in the user table")
```
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/operator_ghpr_e2e_test_kind/detail/operator_ghpr_e2e_test_kind/135/pipeline

PTAL

**Comment by shonge** [comment]
@weekface @tennix  PTAL

File: charts/tidb-cluster/templates/scripts/_initialize_tidb_users.py.tpl [hunk 4]
```diff
@@ -12,12 +13,16 @@ for file in os.listdir(password_dir):
...
     for line in sql.readlines():
         conn.cursor().execute(line)
         conn.commit()
 {{- end }}
+if permit_host != '%%':
+    conn.cursor().execute("update mysql.user set Host=%s where User='root';", (permit_host,))
+    conn.cursor().execute("flush privileges;")
+    conn.commit()
+conn.cursor().execute("flush privileges;")
+conn.commit()
```
Comment (cofyc) [inline charts/tidb-cluster/templates/scripts/_initialize_tidb_users.py.tpl]: why these two lines are duplicated?

**Comment by cofyc** [comment]
@weekface @tennix The reviews were dismissed in last commit. PTAL again.

**Comment by cofyc** [comment]
Thanks for your contribution!

**Pull request merged**